### DISCUSSION TITLE
Kvalitetskrav

### DISCUSSION THREAD

**Discussion by stigbd** [langrenn-sprint/issue-tracker#10] [Ideas]
Tråd som holder på diskusjon rundt kvalitetskrav (før: "ikke-funksjonelle krav").

Vi går for en event drevet arkitektur. (EDA)

Eventer:
- competitior_started: en utøver starter. Kan også trigges av at klokkeslett for oppsatt start passeres
- competitior_finished: en utøver passerer målstreken.
- competitor_intermediate_x: en utøver passerer mellomtid på sensor x

Følgande teknologier bør vurderes:
- eventstreaming: for utveksling av hendelser mellom "sensorer" og server. Eks: https://kafka.apache.org/
- websockets: for "sanntids" oppdateringer knytta til hendelser mellom server og klienter. Eks: https://websockets.readthedocs.io/en/stable/index.html

**Comment by heming-h** [comment]
I vår "vision" verden er disse eventene enten en videosnutt eller et bilde. Nå er det implementer som filer som lagres til en Bucket og tjenestene poller på disse. Tidligere brukte vi også PubSub, men det ble bare ekstra kompleksitet å ha både Bucket og PubSub ettersom det funker ganske greit å polle direkte på en Bucket. Jeg tenker at skal vi endre til for eksempel websockets så må det være fordi det forenkler løsningen vår.

**Reply by stigbd** [reply]
Merk at posten min over var frå 2021. Mykje anna er tenkt sidan då :-)
